File: src/lib-common/salt.js
function salt(x,y)
{
 check(is_str,x)
 
 if(is_undef(y))
  return salt(x,"azertyuiop")
  
 let r=""
 const a1=explode(x)
 const a2=explode(y)
 
 while(is_full(a1))
 {
  if(is_empty(a2))
  {
   const a3=explode(y)
   
   append(a2,a3)
  }
   
  const c1=shift(a1)
  const c2=shift(a2)
  const n1=asc(c1)
  const n2=asc(c2)
  const n=n1^n2
  const c=chr(n)
  
  r=concat(r,c)
 }
 
 return r
}

File: src/lib-common/salt.cs
fn salt x:str y
 if is_undef y
  ret salt x "azertyuiop"

 var r ""
 let a1 explode x
 let a2 explode y

 while is_full a1
  if is_empty a2
   let a explode y

   append a2 a
  end

  let c1 shift a1
  let c2 shift a2
  let n1 asc c1
  let n2 asc c2
  let n xor n1 n2
  let c chr n

  assign r concat r c
 end

 ret r
end
